lib/src/async_import_cache.dart

```diff
@@ -95,20 +96,40 @@ final class AsyncImportCache {
       {Iterable<AsyncImporter>? importers,
       Iterable<String>? loadPaths,
       PackageConfig? packageConfig,
-      Logger? logger})
+      Logger? logger,
+      Set<Deprecation>? fatalDeprecations,
+      Set<Deprecation>? futureDeprecations,
+      Set<Deprecation>? silenceDeprecations})
```

Comment:
I don't really like this change. It's not actually providing any better guarantees—the caller still has to remember to explicitly handle the deprecation options for a new `ImportCache`—and it's reducing the separation of concerns between the way a `Logger` is configured and the way an `ImportCache` is configured. I'd rather just wrap the logger explicitly in the places it needs to be wrapped.

Reply:
Moved the wrapping for import cache into a static method so that it's now always wrapped explicitly, but we don't have to repeat the boilerplate (including the comment about not limiting repetition) in each place we wrap a logger this way.

Reply:
Oh, whoops. I think I accidentally removed them from `ImportCache` instead of `AsyncImportCache` and then that got undone by synchronizing.